ui/src/plugin/PluginDetailView.tsx

```diff
@@ -0,0 +1,259 @@
+                    ) : null}
+                    {this.props.name}
+                    {this.props.refresh ? (
+                        <Button
+                            style={{float: 'right'}}
+                            onClick={() => {
+                                this.props.refresh!();
+                            }}>
+                            <Refresh />
+                        </Button>
+                    ) : null}
+                </h1>
+                {this.props.description ? (
+                    <h4 style={{color: 'grey'}}>{this.props.description}</h4>
+                ) : null}
+                <hr />
+                <div
+                    className={this.props.name
+                        .toLowerCase()
+                        .trim()
+                        .replace(/ /g, '-')}>
+                    {this.props.children}
+                </div>
+            </Container>
+        );
+    }
+}
+
+class ConfigurerPanel extends Component<
+    {pluginInfo: IPlugin; initialConfig: string; save: (newConfig: string) => Promise<void>},
+    {unsavedChanges: string}
+> {
+    public state = {unsavedChanges: ''};
+
+    public render() {
+        return (
+            <div>
+                <MonacoEditor
+                    language="yaml"
+                    theme="vs-dark"
+                    height="15em"
+                    ref="monaco"
+                    value={
+                        this.state.unsavedChanges
+                            ? this.state.unsavedChanges
+                            : this.props.initialConfig
+                    }
+                    onChange={(newConfig) => {
+                        this.setState({unsavedChanges: newConfig});
+                    }}
+                />
+                <br />
+                <Button
+                    variant="contained"
+                    color="primary"
+                    fullWidth={true}
+                    disabled={
+                        !this.state.unsavedChanges ||
+                        this.state.unsavedChanges === this.props.initialConfig
+                    }
+                    className="config-save"
+                    onClick={() => {
+                        let newConfig = ((this.refs.monaco as any).editor as any) // tslint:disable-line
+                            .getModel()
+                            .getValue();
+                        this.props
+                            .save(newConfig)
+                            .then(() => {
+                                this.setState({unsavedChanges: ''});
+                            })
+                            .catch((err) => {
+                                console.error(err);
+                            });
+                    }}>
+                    Save
+                </Button>
+            </div>
+        );
+    }
+}
+
+class DisplayerPanel extends Component<{pluginInfo: IPlugin; displayText: string}> {
```

Comment:
React.Component's with no state can be converted to a StatelessFunctionalComponent, they are a little easier to read IMO.
```typescript
const DisplayerPanel: React.SFC<{displayText: string}> = ({displayText}) => {
    return <ReactMarkDown source={displayText} />;
};
```